### PR TITLE
document non-revenue trips

### PR DESCRIPTION
**Pull request by Whoops**
Asana Task: [[extra] 📝🚝🚫💰 Update GTFS-RT Documentation to explain non-revenue service representation](https://app.asana.com/0/584764604969369/1205761861495467/f)

### CODE REVIEW

**Comment by rymarczy** [comment]
How is this `uncertainty` value different and/or the same compared to the `uncertainty` values specified in the `StopTimeEvent` message? https://gtfs.org/realtime/reference/#message-stoptimeevent

**Comment by Whoops** [comment]
> How is this `uncertainty` value different and/or the same compared to the `uncertainty` values specified in the `StopTimeEvent` message? https://gtfs.org/realtime/reference/#message-stoptimeevent

Whoops, it isn't. I had it on the wrong message. I've updated the PR to fix that!

File: reference/gtfs-realtime.md [hunk 1]
```diff
@@ -141,3 +174,5 @@ Each `multi_carriage_details` element contains a [`CarriageDetails`](https://gtf
 [mtd]: https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#message-tripdescriptor
 [mvd]: https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#message-vehicledescriptor
 [mvp]: https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#message-vehicleposition
+[msr]: https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#enum-schedulerelationship
+[mtse]: https://github.com/google/transit/blob/master/gtfs-realtime/spec/en/reference.md#message-stoptimeevent
```
[the remark this answers is not in the document]
Comment (Whoops) [reply reference/gtfs-realtime.md]: Thanks!

File: reference/gtfs-realtime.md [hunk 2]
```diff
@@ -8,77 +8,94 @@ not replace, [the official GTFS Realtime documentation][rt-docs].
...
+### Uncertainly values
+
+Our [`StopTimeEvent`][mste] messages _may_ have an `uncertainty` value associated
+with them. When present their meanings are as follows.
+
+For Bus trips:
+`uncertainty` | Meaning
+------------- | ------
+\< 300 | Valid real-time prediction
+300 | Real-time prediction not available. This code is primarily used when a vehicle has not yet been assigned to the trip, (i.e. because the block has not started yet). It is a schedule-based prediction, but Swiftly adjusts the schedule-based prediction time using observed historical travel times to make predictions more accurate than the schedule.
```
Comment (fsaid90) [inline reference/gtfs-realtime.md:49]: I don't know that we should mention Swiftly by name here - I'd prefer that we reword to remove references to Swiftly specifically, but tagging @jfabi to get his view as well

File: reference/gtfs-realtime.md [hunk 3]
```diff
@@ -8,77 +8,94 @@ not replace, [the official GTFS Realtime documentation][rt-docs].
...
+301 | Valid real-time prediction, though the bus appears to be stalled or significantly delayed and predictions are not as accurate
+\> 301 | Likely invalid prediction, recommend not showing anything (and not showing scheduled time), very rare situation.
+
+For Rail trips:
```
Comment (fsaid90) [inline reference/gtfs-realtime.md:53]: Do we have uncertainty values for CR predictions?

If we do, we might have to stratify this further (and we'd have to look for Swiftly's definitions)

File: reference/gtfs-realtime.md [hunk 4]
```diff
@@ -133,6 +133,20 @@ Each `multi_carriage_details` element contains a [`CarriageDetails`](https://gtf
 | occupancy_percentage | int32   | Individual car percentage occupancy     |
 | occupancy_status     | int32   | Individual car occupancy status         |
 
+### Non-Revenue trips
+
+Non-revenue vehicle movements, such as travelling to the start of a trip, or movements
+within the yard may be exposed in the enchanced feed as non-revenue trips. In general,
```
Comment (fsaid90) [inline reference/gtfs-realtime.md:156]: Suggest rewording "such as travelling to the start of a trip, or movements within the yard may be exposed" to "such as traveling to the start point of a trip or movements within a yard,"

File: reference/gtfs-realtime.md [hunk 5]
```diff
@@ -133,6 +133,20 @@ Each `multi_carriage_details` element contains a [`CarriageDetails`](https://gtf
...
+these trips should not be shown to riders as they do not accept passengers. These trips
+can be identified by the lack of a `trip_id` within the [`TripDescriptor`][mtd] message
+combined with all [`StopTimeUpdate`][mstu] messages having a [`ScheduleRelationship`][msr]
+of skipped.
+
+Non-revenue trips will only appear in the enhanced feed.
```
Comment (fsaid90) [inline reference/gtfs-realtime.md:162]: Suggest removing "will"

File: reference/gtfs-realtime.md [hunk 6]
```diff
@@ -133,6 +133,20 @@ Each `multi_carriage_details` element contains a [`CarriageDetails`](https://gtf
...
+
+At this time we are working to identify non-revenue light rail trips. In the meantime
+these trips may appear as revenue trips in the feeds.
```
Comment (fsaid90) [inline reference/gtfs-realtime.md:165]: Suggested rewording of 

"At this time we are working to identify non-revenue light rail trips. In the meantime
these trips may appear as revenue trips in the feeds."

to

"At this time, we are continuing to work on identifying non-revenue light rail trips (Green Line, Mattapan Trolley). In the meantime, these trips may appear as revenue trips in the feeds."

File: reference/gtfs-realtime.md (hunk 4 above)
Comment (Whoops) [reply reference/gtfs-realtime.md]: done.

File: reference/gtfs-realtime.md (hunk 2 above)
Comment (Whoops) [reply reference/gtfs-realtime.md]: Agreed. I've rephrased to we.

File: reference/gtfs-realtime.md (hunk 5 above)
Comment (Whoops) [reply reference/gtfs-realtime.md]: done.

File: reference/gtfs-realtime.md (hunk 6 above)
Comment (Whoops) [reply reference/gtfs-realtime.md]: Good idea! Added

File: reference/gtfs-realtime.md [hunk 7]
```diff
@@ -125,14 +145,27 @@ Crowding data is provided within the `multi_carriage_details` element in the `Ve
...
+| carriage_sequence    | int32                    | The order of a carriage within a train  |
+| id                   | string                   | Not used by the MBTA; expect null.      |
+| label                | string                   | Individual car ID                       |
+| occupancy_percentage | int32                    | Individual car percentage occupancy     |
 | occupancy_status     | [`OccupancyStatus`][mos] | Individual car occupancy status         |
 
+### Non-Revenue trips
+
+Non-revenue vehicle movements, such as traveling to the start point of a trip or movements within a yard,
+may be exposed in the enchanced feed as non-revenue trips. In general, these trips should not be shown
```
Comment (meagharty) [inline reference/gtfs-realtime.md:159]: typo: enhanced

File: reference/gtfs-realtime.md [hunk 8]
```diff
@@ -125,14 +145,27 @@ Crowding data is provided within the `multi_carriage_details` element in the `Ve
...
+to riders as they do not accept passengers. These trips can be identified `trip_id`within the
```
Comment (meagharty) [inline reference/gtfs-realtime.md:160]: nit: can we add a space after `trip_id`?

File: reference/gtfs-realtime.md [hunk 9]
```diff
@@ -8,77 +8,97 @@ not replace, [the official GTFS Realtime documentation][rt-docs].
 
 The MBTA provides GTFS Realtime feeds in several formats:
 
-* [Standard](#standard-feeds)
-* [JSON](#json-feeds)
-* [Enhanced JSON](#enhanced-json-feeds)
-
+- [Standard](#standard-feeds)
```
Comment (meagharty) [inline reference/gtfs-realtime.md:11]: It looks like there's some accidental duplication with these links

File: reference/gtfs-realtime.md [hunk 10]
```diff
@@ -8,77 +8,97 @@ not replace, [the official GTFS Realtime documentation][rt-docs].
...
-message | object
-string | string
-int32, int64, uint32, uint64, float, double | number
-enum | string (the name of the value, e.g. `NO_SERVICE`)
-field with cardinality "Many" | array of the specified type
-"Optional" or "Conditionally required" field | key omitted from the object if not provided
-
+| Protobuf                                     | JSON                                              |
+| -------------------------------------------- | ------------------------------------------------- |
+| message                                      | object                                            |
```
Comment (meagharty) [inline reference/gtfs-realtime.md:78]: ty for the fixed spacing in the text rendering 😌 👍

File: reference/gtfs-realtime.md [hunk 11]
```diff
@@ -125,14 +142,27 @@ Crowding data is provided within the `multi_carriage_details` element in the `Ve
...
+| id                   | string                   | Not used by the MBTA; expect null.      |
+| label                | string                   | Individual car ID                       |
+| occupancy_percentage | int32                    | Individual car percentage occupancy     |
 | occupancy_status     | [`OccupancyStatus`][mos] | Individual car occupancy status         |
 
+### Non-Revenue trips
+
+Non-revenue vehicle movements, such as traveling to the start point of a trip or movements within a yard,
+may be exposed in the enhanced feed as non-revenue trips. In general, these trips should not be shown
+to riders as they do not accept passengers. These trips can be identified `trip_id` within the
```
Comment (fsaid90) [inline reference/gtfs-realtime.md:157]: We should probably update this to reflect the actual identification mechanism per the revised revenue TripDescriptor

**Review by fsaid90 — CHANGES_REQUESTED** [review]

**Review by fsaid90 — APPROVED** [review]
LGTM! Thank you 😃
